### ISSUE TITLE
NIOFileSytem: crash when passing `maximumSizeAllowed: .gibibytes(.max)` to disable limits

### ISSUE THREAD

**Issue by weissi** [apple/swift-nio#2877]
```
request.body = try await Data(buffer: ByteBuffer(
    contentsOf: filePath,
    maximumSizeAllowed: .gibibytes(.max)
))
```

crashes with

```
Thread 5 Crashed:: NIO-SGLTN-0-#4
0   0x104f0e510 Swift runtime failure: arithmetic overflow + 0 [inlined]
1   0x104f0e510 static ByteCount.gibibytes(_:) + 76 (ByteCount.swift:79)
```

but it shouldn't.

**Comment by glbrntt** [comment]
Agree. This intersects a bit weirdly with https://github.com/apple/swift-nio/issues/2878 – if `.gigabytes(.max)` is okay, is `.megabytes(.max)` okay? Maybe we should reshape the API and allow `maximumSizeAllowed` to be `nil` _or_ add a sentinel value to the `ByteCount` which makes the intent clearer (`.max`, maybe?).

**Comment by weissi** [comment]
> Agree. This intersects a bit weirdly with #2878 – if `.gigabytes(.max)` is okay, is `.megabytes(.max)` okay? Maybe we should reshape the API and allow `maximumSizeAllowed` to be `nil` _or_ add a sentinel value to the `ByteCount` which makes the intent clearer (`.max`, maybe?).

Yes, I'd suggest just `.unlimitedDangerousForSecurityReasons` or something straight on `ByteCount`.

**Comment by glbrntt** [comment]
Feels weird having it on `ByteCount` but it might just be the pragmatic choice.

**Comment by weissi** [comment]
> Feels weird having it on `ByteCount` but it might just be the pragmatic choice.

Why? `ByteCount.unlimited` sounds alright, no?
